### PR TITLE
Ease up setting affinity on linux x86_64

### PR DESCRIPTION
**Pull request by blazember**
Setting the thread affinity with the OpenHFT library requires the
OpenHFT lib to be on the classpath and libffi.so to be loadable too.
This makes the users forced to configure the system to use the affinity
feature. With this change we make this easy for the users running
Hazelcast on Linux x86_64. If not explicitly disabled with
`-Dhazelcast.affinity.lib.disabled=true`, we load our
`libaffinity_helper.so` that we bundle in our OS and EE jar instead of
using the OpenHFT library, therefore, enabling thread affinity on Linux
x86_64 requires no more than just setting the system arguments.

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2142/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2142/Hazelcast-pr-builder-2142.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/hazelcast/src/main/java/com/hazelcast/internal/util/ThreadAffinityHelper.java:47: Class ThreadAffinityHelper should be declared as final. [FinalClass]
--------------------------
</details>
</pre>

**Review by pveentjer — APPROVED** [review]

**Comment by mmedenjak** [comment]
I guess you want to remove it from `pom.xml` as well - https://github.com/hazelcast/hazelcast/blob/da71f523a6df8cef96c87045c38f6c6cba23cca1/hazelcast/pom.xml#L319-L326

File: hazelcast/src/main/resources/affinity_helper.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Follow the steps below to compile the shared library for Linux #
+
+`PATH_TO_JDK_INCLUDE_DIR`: The full path for the include directory under you JDK installation.
```
Comment (mmedenjak) [inline hazelcast/src/main/resources/affinity_helper.md:3]: Minor: you -> your

**Review by mmedenjak — APPROVED** [review]

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2262/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2262/Hazelcast-pr-builder-2262.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   FrozenPartitionTableTest.partitionTable_shouldBeFixed_whenMemberRestarts_usingNewUuid:231 ? Completion
[INFO] 
[ERROR] Tests run: 31485, Failures: 0, Errors: 1, Skipped: 979
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by blazember** [comment]
Merging with unrelated failure. Created https://github.com/hazelcast/hazelcast/issues/17505 for the new failure.

**Pull request merged**